### PR TITLE
[Tests] Increase dockerized tests timeout

### PR DESCRIPTION
**Pull request by yaelgen**
fails locally, increasing the timeout solves the problem.

### CODE REVIEW

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
I am not sure this should be the solution as it can cause us miss possible decreases in performance.

Possible solution I see is resolving the timeout based on the system we are running on, if for example we run in github actions leave it as 30 and if not running in actions set to 90, as on personal computer the timeout can be differ due to poor network connection etc.

**Comment by Tankilevitch** [comment]
> I am not sure this should be the solution as it can cause us miss possible decreases in performance.
> 
> Possible solution I see is resolving the timeout based on the system we are running on, if for example we run in github actions leave it as 30 and if not running in actions set to 90, as on personal computer the timeout can be differ due to poor network connection etc.

Discussed with @quaark offline, decided that it is ok to increase the timeout so developers on apple silicon won't fail on the integration tests as now it takes longer to load due to the qume emulator, and in addition @yaelgen will add a test that actually tests the desired timeout (30 sec) like we had before

**Comment by quaark** [comment]
> > I am not sure this should be the solution as it can cause us miss possible decreases in performance.
> > Possible solution I see is resolving the timeout based on the system we are running on, if for example we run in github actions leave it as 30 and if not running in actions set to 90, as on personal computer the timeout can be differ due to poor network connection etc.
> 
> Discussed with @quaark offline, decided that it is ok to increase the timeout so developers on apple silicon won't fail on the integration tests as now it takes longer to load due to the qume emulator, and in addition @yaelgen will add a test that actually tests the desired timeout (30 sec) like we had before

I'll clarify, @yaelgen please add a test that will only test the api boot up speed (30 seconds). Add a comment explaining it'll probably fail on an Apple Silicon Mac, but passes on native x86 architecture.

File: tests/rundb/test_httpdb.py [hunk 1]
```diff
@@ -201,6 +203,24 @@ def test_log(create_server):
     assert data == body, "bad log data"
 
 
+@pytest.mark.skipif(
+    sys.platform == "darwin",
+    reason="We are developing on Apple Silicon Macs,"
+    " which will most likely fail this test due to the qume emulator,"
```
Comment (liranbg) [inline tests/rundb/test_httpdb.py:209]:
```suggestion
    " which will most likely fail this test due to the qemu being slow,"
```

File: tests/rundb/test_httpdb.py [hunk 2]
```diff
@@ -201,6 +203,24 @@ def test_log(create_server):
...
+    " but should pass on native x86 architecture",
```
Comment (liranbg) [inline tests/rundb/test_httpdb.py:210]:
```suggestion
    " but should pass on native architecture",
```

File: tests/rundb/test_httpdb.py [hunk 3]
```diff
@@ -201,6 +203,24 @@ def test_log(create_server):
...
+)
+def test_api_boot_speed(create_server):
+    run_times = 50
+    expected_time = 30
+    runs = []
+    for i in range(run_times):
```
Comment (liranbg) [inline tests/rundb/test_httpdb.py:216]:
```suggestion
    for _ in range(run_times):
```

File: tests/rundb/test_httpdb.py [hunk 4]
```diff
@@ -201,6 +203,24 @@ def test_log(create_server):
...
+        start_time = time.perf_counter()
+        create_server()
+        end_time = time.perf_counter()
+        runs.append(end_time - start_time)
+    assert sum(runs) / run_times <= expected_time
```
Comment (liranbg) [inline tests/rundb/test_httpdb.py:221]:
```suggestion
    avg_run_time = sum(runs) / run_times
    assert avg_run_time <= expected_time, "Seems like a performance hit on creating api server"
```

Comment (liranbg) [reply tests/rundb/test_httpdb.py:221]: better explain _why_ the assertion failed - throw a bone to whoever fails on it
in addition, better save the magic calculations on a side variable to give it a name of what we calculate

**Review by liranbg — CHANGES_REQUESTED** [review]

**Pull request merged**